### PR TITLE
7 d ints and reals

### PR DESCRIPTION
**Pull request by aarograh**
Adds support for 5D, 6D, and 7D ints and reals in parameter lists with testing.  Also adds support to read and write said datasets as part of a parameter list in the HDF5 file.

### CODE REVIEW

File: unit_tests/testParameterLists/testParameterLists.f90 [hunk 1]
```diff
@@ -4653,6 +4677,2790 @@ SUBROUTINE testSDKa4()
...
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat')
+
+  COMPONENT_TEST('%edit(...)')
+  CALL testParam%init('testSNKa5',valsnka5,'The numbers 5, 7, 6, & 8')
+  CALL testParam%edit(OUTPUT_UNIT,0)
+  CALL testParam%clear()
+  CALL testParam%init('testSNKa5',valsnka5)
+  CALL testParam%edit(OUTPUT_UNIT,0)
+  CALL testParam%clear()
+  CALL testParam%edit(OUTPUT_UNIT,0)
```
Comment (HendersonSC) [inline unit_tests/testParameterLists/testParameterLists.f90:4738]: What is this component test even doing?

**Review by HendersonSC — COMMENTED** [review]
Okay...I am hard pressed to say this looks goo, but that is more a statement to the limitations of adding such features than the actual code. Left a few comments in the unit test, they are in no way really limiting and due to the size of the unit test might be better to spin off into a separate issue.

File: unit_tests/testParameterLists/testParameterLists.f90 [hunk 2]
```diff
@@ -4653,6 +4677,2790 @@ SUBROUTINE testSDKa4()
...
+  valsnka5(2,2,1,:,1)=4_SNK
+  valsnka5(1,1,2,:,1)=5_SNK
+  valsnka5(2,1,2,:,1)=6_SNK
+  valsnka5(1,2,2,:,1)=7_SNK
+  valsnka5(2,2,2,:,1)=8_SNK
+  !test init
+
+  COMPONENT_TEST('%init(...)')
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat 1')
+  ASSERT(LEN(testParam%name) == 0,'%name 1')
```
[the remark this answers is not in the document]
Comment (aarograh) [reply unit_tests/testParameterLists/testParameterLists.f90:4697]: 100% agreed.  The first time it ran I only had 5 failures and they would have been much easier to track down had ASSERT_EQ been used... I'll make an issue for this (should be easy search and replace but should be done on its own)

File: unit_tests/testParameterLists/testParameterLists.f90 (hunk 1 above)
Comment (aarograh) [reply unit_tests/testParameterLists/testParameterLists.f90:4738]: I guess just making sure it doesn't bomb hard while writing.  You're silly if you think I actually read any of this code though.

File: unit_tests/testParameterLists/testParameterLists.f90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (aarograh) [reply unit_tests/testParameterLists/testParameterLists.f90:4697]: Issue #312 addresses this

**Review by HendersonSC — APPROVED** [review]

**Comment by aarograh** [comment]
Only test failure is unrelated

**Pull request merged**